examples/utils/oauth.js

```diff
@@ -0,0 +1,21 @@
+// Authenticate with PureCloud
+function authenticate(client, pcEnvironment) {
+    /*
+    * Note: To use this app in your own org, you will need to create your own OAuth2 Client(s)
+    * in your PureCloud org.  After creating the Implicit grant client, map the client id(s) to
+    * the specified region key(s) in the object below, deploy the page, and configure an app to point to that URL.
+    */
+    const pcOAuthClientIds = {'mypurecloud.com': 'implicit-oauth-client-id-here'};
+    const clientId = pcOAuthClientIds[pcEnvironment];
+    if (!clientId) {
+        return Promise.reject(new Error(pcEnvironment + ': Unknown/Unsupported PureCloud Environment'));
+    }
+
+    const { origin, protocol, host, pathname } = window.location;
+    const redirectUrl = (origin || `${protocol}//${host}`) + pathname;
+
+    return client.loginImplicitGrant(clientId, redirectUrl, {state: `pcEnvironment=${pcEnvironment}`})
+        .then(data => {
+            window.history.replaceState(null, null, `${pathname}?${data.state}`);
```

Comment:
mdn suggests using '' for the title as opposed to null for future proofing the call across browsers.  But, I haven't used history state much, so take this with a grain of salt if null is more appropriate.

Reply:
Also, nice approach using replaceState.  Clean, simple, addresses the issue 👍